core/whistle_apps-1.0.0/src/whapps_speech.erl

```diff
@@ -138,9 +138,70 @@ create(<<"ispeech">> = Engine, Text, Voice, Format, Options) ->
+    VoiceMappings = [ %% Владимир8000
+                     {<<"male/ru-ru">>, <<208,146,208,187,208,176,208,180,208,184,208,188,208,184,209,128,56,48,48,48>>}
+                     %% Юлия8000
+                     ,{<<"female/ru-ru">>, <<208,174,208,187,208,184,209,143,56,48,48,48>>}
+                      %% Владимир8000
+                     ,{<<"male/ru-ru/vladimir">>, <<208,146,208,187,208,176,208,180,208,184,208,188,208,184,209,128,56,48,48,48>>}
+                     %% Юлия8000
+                     ,{<<"female/ru-ru/julia">>, <<208,174,208,187,208,184,209,143,56,48,48,48>>}
+                     %% Анна8000
+                     ,{<<"female/ru-ru/anna">>, <<208,144,208,189,208,189,208,176,56,48,48,48>>}
+                     %% Виктория8000
+                     ,{<<"female/ru-ru/viktoria">>, <<208,146,208,184,208,186,209,130,208,190,209,128,208,184,209,143,56,48,48,48>>}
+                     %% Александр8000
+                     ,{<<"male/ru-ru/alexander">>, <<208,144,208,187,208,181,208,186,209,129,208,176,208,189,208,180,209,128,56,48,48,48>>}
+                     %% Мария8000
+                     ,{<<"female/ru-ru/maria">>, <<208,156,208,176,209,128,208,184,209,143,56,48,48,48>>}
+                     %% Лидия8000
+                     ,{<<"female/ru-ru/lidia">>, <<208,155,208,184,208,180,208,184,209,143,56,48,48,48>>}
+                    ],
+    case props:get_value(wh_util:to_lower_binary(Voice), VoiceMappings) of
+        'undefined' ->
+            {'error', 'invalid_voice'};
+        VFabricVoice ->
+            BaseUrl = whapps_config:get_string(?MOD_CONFIG_CAT, <<"tts_url">>, <<"https://voicefabric.ru/WSServer/ws/tts">>),
+            ApiKey = whapps_config:get_binary(?MOD_CONFIG_CAT, <<"tts_api_key">>, <<>>),
+            Data = [{<<"apikey">>, ApiKey}
+                    ,{<<"ttsVoice">>, VFabricVoice}
+                    ,{<<"textFormat">>, <<"text/plain">>}
+                    ,{<<"text">>, Text}
+                   ],
+            ArgsEncode = whapps_config:get_binary(?MOD_CONFIG_CAT, <<"tts_args_encode">>, <<"urlencode">>),
+            {Headers, Body} = voicefabric_request_body(ArgsEncode, Data),
+            HTTPOptions = [{'response_format', 'binary'} | Options],
+            Response = ibrowse:send_req(BaseUrl, Headers, 'post', Body, HTTPOptions),
+            create_response(Engine, Response)
+    end;
 create(_, _, _, _, _) ->
     {'error', 'unknown_provider'}.
 
+voicefabric_request_body(<<"urlencode">>, Data) ->
+    Headers_ = [{"Content-Type", "application/x-www-form-urlencoded"}],
+    <<$&, Body_/binary>> = iolist_to_binary([[$&, Key, $=, Val] || {Key, Val} <- Data]),
+    {Headers_, Body_};
+voicefabric_request_body(<<"multipart">>, Data) ->
+    Boundary = iolist_to_binary([<<"--boundary--">>
+                                 ,couch_mgr:get_uuid(),
+                                 <<"--boundary--">>
+                                ]),
+    Headers_ = [{"Content-Type"
```

Comment:
I'm curious why you suffix the Variables with the '_'. When prefixed, the compiler won't complain about unused vars but as far as I know, '_' as a suffix does nothing special.